### PR TITLE
Trev8939/viewshed

### PR DESCRIPTION
**Pull request by TADraeseke**
Ready for your review @doneill.

Can you have a look at the creation of the FeatureCollectionTable? All other platforms appear to be able to pass an empty list of Fields as the first parameter in the FeatureCollectionTable constructor. I could not replicate that behaviour on Android, so have a work around in place at the moment. See the //TODO.

Thanks!

### CODE REVIEW

File: viewshed-geoprocessing/src/main/java/com/esri/arcgisruntime/sample/viewshedgeoprocessing/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    if (mGeoprocessingJob != null) {
+      mGeoprocessingJob.cancel();
+    }
+
+    // TODO - This feels like a bit of a hack. Using new ArrayList<Field>() as the first parameter of new
+    // TODO - FeatureCollectionTable(new ArrayList<Field>(), ...) throws an error
+    Field field = Field.createInteger("placeholder", null);
+
+    // create feature collection table for point geometry
+    final FeatureCollectionTable featureCollectionTable = new FeatureCollectionTable(Collections.singletonList(field),
```
Comment (doneill) [inline viewshed-geoprocessing/src/main/java/com/esri/arcgisruntime/sample/viewshedgeoprocessing/MainActivity.java]: Developer pattern should be to create a the table schema by defining the a set of fields:

```java
List<Field> fields = new ArrayList<>(1);
// create field with same alias as name
Field field = Field.createString("observer", "", 8);
fields.add(field);
final FeatureCollectionTable featureCollectionTable = new FeatureCollectionTable(fields,
    GeometryType.POINT, point.getSpatialReference());
```

File: viewshed-geoprocessing/src/main/java/com/esri/arcgisruntime/sample/viewshedgeoprocessing/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+            @Override public void run() {
+              GeoprocessingResult geoprocessingResult = mGeoprocessingJob.getResult();
+              GeoprocessingFeatures resultFeatures = (GeoprocessingFeatures) geoprocessingResult.getOutputs()
+                  .get("Viewshed_Result");
+              FeatureSet featureSet = resultFeatures.getFeatures();
+              for (Feature feature : featureSet) {
+                Graphic graphic = new Graphic(feature.getGeometry());
+                mResultGraphicsOverlay.getGraphics().add(graphic);
+              }
+            }
```
Comment (doneill) [inline viewshed-geoprocessing/src/main/java/com/esri/arcgisruntime/sample/viewshedgeoprocessing/MainActivity.java]: You need to check to ensure job returns result: 

```java
GeoprocessingResult geoprocessingResult = mGeoprocessingJob.getResult();
  if(geoprocessingResult != null){
      // get and iterate over graphic features
  }else{
      Toast.makeText(getApplicationContext(), "Geoprocessing result failed, try again", Toast.LENGTH_SHORT).show();
  }
```

**Comment by TADraeseke** [comment]
Thanks @doneill. PR fixes in place!

**Comment by doneill** [comment]
@TADraeseke looks good, merging.

**Pull request merged**